src/lcm/create_params.py

```diff
@@ -1,138 +1,176 @@
 
-    for var in stochastic_variables:
-        if (
-            not variable_info.loc[var, "is_state"]
-            or not variable_info.loc[var, "is_discrete"]
-        ):
-            raise ValueError(
-                f"Shock {var} is stochastic but not a discrete state variable.",
-            )
-
-    params = {}
-    for var in stochastic_variables:
-        # read signature of next function corresponding to stochastic variable
-        dependencies = list(
-            inspect.signature(model["functions"][f"next_{var}"]).parameters,
-        )
+    # ----------------------------------------------------------------------------------
+    # Assert that all stochastic variables are discrete state variables
+    # ----------------------------------------------------------------------------------
+    discrete_state_vars = set(variable_info.query("is_state & is_discrete").index)
 
-        _check_variables_are_all_discrete_or_period(
-            variables=dependencies,
-            variable_info=variable_info,
-            msg_suffix=(
-                f"The function next_{var} can only depend on discrete variables or"
-                f"'_period'."
-            ),
-        )
+    invalid = set(stochastic_variables) - discrete_state_vars
 
-        # get dimensions of variables that influence the stochastic variable
-        dimensions = [
-            len(grids[dep]) if dep != "_period" else model["n_periods"]
-            for dep in dependencies
-        ]
-        # add dimension of stochastic variable to last axis
-        dimensions = (*dimensions, len(grids[var]))
+    if invalid:
+        raise ValueError(
+            f"The following variables are stochastic, but are not discrete state "
+            f"variables: {invalid}. This is currently not supported.",
+        )
 
-        params[var] = np.full(dimensions, np.nan)
+    # ----------------------------------------------------------------------------------
+    # Create template matrices for stochastic transitions
+    # ----------------------------------------------------------------------------------
 
-    return params
+    # Stochastic transition functions can only depend on discrete vars or '_period'.
+    valid_vars = set(variable_info.query("is_discrete").index) | {"_period"}
 
+    stochastic_transition_params = {}
+    invalid_dependencies = {}
 
-def _create_standard_params():
-    return {"beta": np.nan}
+    for var in stochastic_variables:
 
+        # Retrieve corresponding next function and its arguments
+        next_var = model_spec["functions"][f"next_{var}"]
+        dependencies = list(inspect.signature(next_var).parameters)
+
+        invalid = set(dependencies) - valid_vars
+        # If there are invalid dependencies, store them in a dictionary and continue
+        # with the next variable to collect as many invalid arguments as possible.
+        if invalid:
+            invalid_dependencies[var] = invalid
+            continue
```

Comment:
I kinda prefer if/else and adding the comment inside the if-statement instead of this style, but I won't argue over it.